docs/guides/workflows_patterns/deployment_airflow.rst

```diff
@@ -1,17 +1,109 @@
-
-#################################
 Deploying Great Expectations with Airflow
-#################################
+=========================================
+
+This guide will help you deploy Great Expectations within an Airflow pipeline. You can use Great Expectations to automate validation of data integrity and navigate your DAG based on the output of validations.
+
+.. admonition:: Prerequisites: This workflow pattern assumes you have already:
+
+    - Configured a :ref:`Data Context <tutorials__getting_started__initialize_a_data_context>`.
+    - Configured an :ref:`Expectations Suite <tutorials__getting_started__create_your_first_expectations>`.
+    - Optional: Configured a :ref:`Checkpoint <tutorials__getting_started__set_up_your_first_checkpoint>`.
+    - Configured an Airflow pipeline (DAG).
+
+There are two supported methods: using an Airflow ``PythonOperator`` to run Validations using python code or invoking the Great Expectations CLI to run a Checkpoint using an Airflow ``BashOperator``.
+
+.. note::
+    There is not currently a Great Expectations custom Airflow operator, however we have heard from users who have implemented their own. We love community contributions so please share your operators!
+
+Check out these links for some examples of Airflow pipelines with Great Expectations Validations ensuring that downstream tasks are protected from upstream issues:
+
+- `Great Expectations Pipeline Tutorial <https://github.com/superconductive/ge_tutorials>`_
+- `Great Expectations / Airflow Demo <https://github.com/superconductive/airflow_meetup_demo>`_
+
+In the first link and the diagram below, you can see a common pattern of using Validation tasks to ensure that the data flowing from one task to the next is correct, and alert the team if it is not. Another common pattern is to branch and change your DAG based on a Validation (e.g. send data for more cleaning before moving to the next task, store it for a postmortem, etc.).
+
+.. image:: ge_tutorials_pipeline.png
+    :width: 800
+    :alt: Airflow pipeline from Great Expectations tutorials repository.
+
+
+We will now explain the two supported methods for using Great Expectations within an Airflow pipeline.
+
+Running a Validation using a ``PythonOperator``
+-----------------------------------------------
+
+1. **Create Validation Methods**
+
+    Create the methods to validate data that will be called in your DAG. In this example our data is contained in a file.
+
+.. code-block:: python
+
+    from airflow import AirflowException
+    from airflow.operators.python_operator import PythonOperator
+    import great_expectations as ge
+
+    ...
+
+    def validate_data(ds, **kwargs):
+
+        # Retrieve your data context
+        context = ge.data_context.DataContext(<insert path to your great_expectations.yml>)
+
+        # Create your batch_kwargs
+        batch_kwargs_file = {
+            "path": <insert path to your data file>,
+            "datasource": "input_files"}
```

Comment:
I want to be clear that the datasource (in this context) isn't defining files; that's coming from the path.

```suggestion
            "datasource": "my_pandas_datasource"}
```